pudl/load.py

```diff
@@ -71,20 +83,85 @@ def _fix_int_cols(table_to_fix,
                 transformed_dct[table_to_fix][column])
 
 
+class BulkCopy(contextlib.AbstractContextManager):
```

Comment:
If I understand correctly, there's an implicit requirement that all the dataframes within any bulk copy would need to contain records from the same database table, as they're being concatenated together eventually, when they exceed the buffer size. So it's for chunking up the larger database insertions, rather than (say) aggregating several smaller heterogeneous dataframes together for insertion. Just might want to make that explicit in the docstring, and add an assert / raise an exception that causes the `add()` method to fail if a dataframe that doesn't match is added.